modules/mailchimp_ecommerce_commerce/mailchimp_ecommerce_commerce.module

```diff
@@ -953,3 +958,100 @@ function _mailchimp_ecommerce_commerce_get_inventory($product) {
 
   return $inventory_amount;
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function mailchimp_ecommerce_commerce_form_mailchimp_ecommerce_admin_sync_orders_alter(&$form, &$form_state) {
+  $form['#submit'][] = 'mailchimp_ecommerce_commerce_admin_sync_orders_submit';
+}
+
+/**
+ * Submit handler for the MailChimp eCommerce sync form.
+ */
+function mailchimp_ecommerce_commerce_admin_sync_orders_submit($form, &$form_state) {
+  if (!empty($form_state['values']['sync_orders'])) {
+    $batch = [
+      'title' => t('Adding orders to MailChimp'),
+      'operations' => [],
+    ];
+
+    // get orders within specified time period (months)
+    $months = abs(intval($form_state['values']['timespan'])) * -1;
+    $min_timestamp = strtotime($months . ' months');
+
+    $query = new EntityFieldQuery;
+    $results = $query
+      ->entityCondition('entity_type','commerce_order')
+      //->propertyCondition('created', $min_timestamp, '>')
+      ->execute();
```

Comment:
I might restrict this to post-checkout states, otherwise they'll be considered Carts:
`$post_checkout_states = array_keys(commerce_order_statuses(array('state' => 'completed')));`

Then:
`$query->propertyCondition('status', $post_checkout_states);`

Oughta do it.

Reply:
my queue processor splits them out into carts or orders based on status down lower in the code.

Reply:
Should add a flag to skip carts.

Reply:
Yup just saw that. I think that'd work too.